chain/client/src/client.rs

```diff
@@ -175,12 +179,22 @@ impl Client {
         next_height: BlockHeight,
         elapsed_since_last_block: Duration,
     ) -> Result<Option<Block>, Error> {
+        #[allow(unused_assignments)]
+        #[allow(unused_mut)]
+        let mut adv_produce_blocks = false;
+
+        #[cfg(feature = "adversarial")]
+        {
+            adv_produce_blocks = self.adv_produce_blocks;
+        }
+
         // Check that this block height is not known yet.
-        if !self.adv_produce_blocks {
+        if !adv_produce_blocks {
```

Comment:
Should we just feature-gate the whole block instead of doing this dancing with the local variable?